### PR TITLE
Replace assert by  is_viur_datastore_request_ok in `transport.pyx`

### LINKED ISSUE #34: `AssertionError` in transport.pyx with no further information
With current viur-datastore 1.3.11, I'm getting this error
```
[2023-09-25 18:39:29,050] /.../viur/core/request.py:320 [ERROR] ViUR has caught an unhandled exception!
[2023-09-25 18:39:29,050] /.../viur/core/request.py:321 [ERROR] 
Traceback (most recent call last):
  File "~/viur-project/.venv/lib/python3.11/site-packages/viur/core/request.py", line 295, in _process
    self._route(path)
  File "~/viur-project/.venv/lib/python3.11/site-packages/viur/core/request.py", line 539, in _route
    res = caller(*self.args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/viur-project/.venv/lib/python3.11/site-packages/viur/core/module.py", line 299, in __call__
    return self._func(self._instance, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/viur-project/deploy/modules/user.py", line 214, in edit
    if not securitykey.validate(skey):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/viur-project/.venv/lib/python3.11/site-packages/viur/core/securitykey.py", line 80, in validate
    if not key or not (entity := db.Get(db.Key(SECURITYKEY_KINDNAME, key))):
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "src/viur/datastore/transport.pyx", line 614, in viur.datastore.transport.Get
AssertionError
```
which relates to the `assert req.status_code == 200` line in below context
```python
    while keyList:
        requestedKeys = keyList[:300]
        postData = {
            "readOptions": readOptions, 
            "keys": requestedKeys, 
        } 
        req = authenticatedRequest( 
            url="https://datastore.googleapis.com/v1/projects/%s:lookup" % projectID, 
            data=json.dumps(postData).encode("UTF-8"), 
        ) 
        assert req.status_code == 200
        assert PyBytes_AsStringAndSize(req.content, &data_ptr, &pysize) != -1
        element = parser.parse(data_ptr, pysize, 1)
        if (element.at_pointer("/found").error() == SUCCESS):
            res.update(toEntityStructure(element.at_key("found"), isInitial=True))
        if (element.at_pointer("/deferred").error() == SUCCESS):
            keyList = toPythonStructure(element.at_key("deferred")) + keyList[300:]
        else:
            keyList = keyList[300:]
```
This is absolutely not useful - why isn't the status_code printed, at least in the AssertionError, to provide a hint about what the problem is?

### PR DESCRIPTION
**Pull request by ArneGudermann**
Resolves  #34

### CODE REVIEW

File: src/viur/datastore/errors.py [hunk 1]
```diff
@@ -168,17 +168,23 @@ class NoMutationResultsError(ViurDatastoreError):
...
+	if resp.status_code != 200:
+		try:
+			error_data = json.loads(resp.content)["error"]
+			if error_data["status"] in conf["verbose_error_codes"]:
+				pprint(error_data)
+			raise CANONICAL_ERROR_CODE_MAP.get(error_data["status"], ViurDatastoreError)(error_data["message"])
+		except ViurDatastoreError:
+			raise
+		except:
+			pprint(f"{resp.url} failed with Code: {resp.content}\nAnd Content:\n{resp.content}")
```
Comment (phorward) [inline src/viur/datastore/errors.py:188]: Is `pprint` here really the best way to chose?

File: src/viur/datastore/transport.pyx [hunk 2]
```diff
@@ -845,7 +847,7 @@ def RunInTransaction(callback: callable, *args, **kwargs) -> Any:
 							if innerArrayElem.at_pointer("/key").error() == SUCCESS:  # We got a new key assigned
 								affectedEntity = currentTxn["affectedEntities"][idx]
 								if not affectedEntity:
-									print(req.content)
+									print(resp.content)
 									raise ViurDatastoreError("Received an unexpected key-update")
```
Comment (phorward) [inline src/viur/datastore/transport.pyx:851]: Don't use print!

Either
```suggestion
									logging.error(f"{resp.content=}")
									raise ViurDatastoreError("Received an unexpected key-update")
```
or
```suggestion
									raise ViurDatastoreError(f"Received an unexpected key-update: {resp.content=}")
```

**Review by phorward — CHANGES_REQUESTED** [review]

File: src/viur/datastore/errors.py (hunk 1 above)
Comment (skoegl) [reply src/viur/datastore/errors.py:188]: I would also prefer logging infra

File: src/viur/datastore/transport.pyx [hunk 3]
```diff
@@ -689,19 +691,19 @@ def Delete(keys: Union[Key, List[Key], Entity, List[Entity]]) -> None:
...
 		data=json.dumps(postData).encode("UTF-8"),
 	)
-	if is_viur_datastore_request_ok(req):
-		assert PyBytes_AsStringAndSize(req.content, &data_ptr, &pysize) != -1
+	if is_viur_datastore_request_ok(resp):
+		assert PyBytes_AsStringAndSize(resp.content, &data_ptr, &pysize) != -1
 		element = parser.parse(data_ptr, pysize, 1)
 		if (element.at_pointer("/mutationResults").error() != SUCCESS):
-			pprint(req.content)
+			pprint(resp.content)
```
Comment (skoegl) [inline src/viur/datastore/transport.pyx:702]: l would prefer appropriate logging.* calls over print() and pprint

File: src/viur/datastore/transport.pyx [hunk 4]
```diff
@@ -689,19 +691,19 @@ def Delete(keys: Union[Key, List[Key], Entity, List[Entity]]) -> None:
...
 			raise NoMutationResultsError("No mutation results received")
 		arrayElem = element.at_key("mutationResults").get_array()
 		if arrayElem.size() != abs(len(keys)):
-			print(req.content)
+			print(resp.content)
```
Comment (skoegl) [inline src/viur/datastore/transport.pyx:706]: l would prefer appropriate logging.* calls over print() and pprint

File: src/viur/datastore/transport.pyx [hunk 5]
```diff
@@ -740,20 +742,20 @@ def Put(entities: Union[Entity, List[Entity]]) -> Union[Entity, List[Entity]]:
...
 	)
 
-	if is_viur_datastore_request_ok(req):
-		assert PyBytes_AsStringAndSize(req.content, &data_ptr, &pysize) != -1
+	if is_viur_datastore_request_ok(resp):
+		assert PyBytes_AsStringAndSize(resp.content, &data_ptr, &pysize) != -1
 		element = parser.parse(data_ptr, pysize, 1)
 		if (element.at_pointer("/mutationResults").error() != SUCCESS):
-			print(req.content)
+			print(resp.content)
```
Comment (skoegl) [inline src/viur/datastore/transport.pyx:754]: l would prefer appropriate logging.* calls over print() and pprint

File: src/viur/datastore/transport.pyx [hunk 6]
```diff
@@ -740,20 +742,20 @@ def Put(entities: Union[Entity, List[Entity]]) -> Union[Entity, List[Entity]]:
...
 			raise ValueError("No mutation-results received")
 		arrayElem = element.at_key("mutationResults").get_array()
 		if arrayElem.size() != abs(len(entities)):
-			pprint(req.content)
+			pprint(resp.content)
```
Comment (skoegl) [inline src/viur/datastore/transport.pyx:758]: l would prefer appropriate logging.* calls over print() and pprint

File: src/viur/datastore/transport.pyx [hunk 7]
```diff
@@ -823,20 +825,20 @@ def RunInTransaction(callback: callable, *args, **kwargs) -> Any:
...
 						)
 
-						is_viur_datastore_request_ok(req)
-						assert PyBytes_AsStringAndSize(req.content, &data_ptr, &pysize) != -1
+						is_viur_datastore_request_ok(resp)
+						assert PyBytes_AsStringAndSize(resp.content, &data_ptr, &pysize) != -1
 						element = parser.parse(data_ptr, pysize, 1)
 						if (element.at_pointer("/mutationResults").error() != SUCCESS):
-							pprint(req.content)
+							pprint(resp.content)
```
Comment (skoegl) [inline src/viur/datastore/transport.pyx:837]: l would prefer appropriate logging.* calls over print() and pprint

File: src/viur/datastore/transport.pyx [hunk 8]
```diff
@@ -823,20 +825,20 @@ def RunInTransaction(callback: callable, *args, **kwargs) -> Any:
...
 							raise NoMutationResultsError("No mutation-results received")
 						arrayElem = element.at_key("mutationResults").get_array()
 						if arrayElem.size() != abs(len(currentTxn["affectedEntities"])):
-							print(req.content)
+							print(resp.content)
```
Comment (skoegl) [inline src/viur/datastore/transport.pyx:841]: l would prefer appropriate logging.* calls over print() and pprint

**Review by skoegl — CHANGES_REQUESTED** [review]
I think, it's better to use logging or loggers - an additional pprint is ok, if we need the output preformatted

**Comment by sveneberth** [comment]
can we please get this ready for release asap?

If we'd already had it in, I would probably have noticed the error from #53 much sooner and I wouldn't have wasted a lot of man-hours troubleshooting. Apart from wasting money on failed tasks.

**Pull request merged**